src/lib.rs

```diff
@@ -324,52 +312,86 @@ mod tests {
         let mime_db = load_test_data();
 
         assert_eq!(
-            mime_db.unalias_mime_type("application/ics"),
-            Some("text/calendar".to_string())
+            mime_db.unalias_mime_type(&Mime::from_str("application/ics").unwrap()),
+            Some(Mime::from_str("text/calendar").unwrap())
         );
-        assert_eq!(mime_db.unalias_mime_type("text/plain"), None);
+        assert_eq!(mime_db.unalias_mime_type(&Mime::from_str("text/plain").unwrap()), None);
     }
 
     #[test]
     fn mime_type_equal() {
         let mime_db = load_test_data();
 
         assert_eq!(
-            mime_db.mime_type_equal("application/wordperfect", "application/vnd.wordperfect"),
+            mime_db.mime_type_equal(
+                &Mime::from_str("application/wordperfect").unwrap(),
+                &Mime::from_str("application/vnd.wordperfect").unwrap(),
+            ),
             true
         );
         assert_eq!(
-            mime_db.mime_type_equal("application/x-gnome-app-info", "application/x-desktop"),
+            mime_db.mime_type_equal(
+                &Mime::from_str("application/x-gnome-app-info").unwrap(),
+                &Mime::from_str("application/x-desktop").unwrap(),
+            ),
             true
         );
         assert_eq!(
-            mime_db.mime_type_equal("application/x-wordperfect", "application/vnd.wordperfect"),
+            mime_db.mime_type_equal(
+                &Mime::from_str("application/x-wordperfect").unwrap(),
+                &Mime::from_str("application/vnd.wordperfect").unwrap(),
+            ),
             true
         );
         assert_eq!(
-            mime_db.mime_type_equal("application/x-wordperfect", "audio/x-midi"),
+            mime_db.mime_type_equal(
+                &Mime::from_str("application/x-wordperfect").unwrap(),
+                &Mime::from_str("audio/x-midi").unwrap(),
+            ),
+            false
+        );
+        assert_eq!(
+            mime_db.mime_type_equal(
+                &Mime::from_str("application/octet-stream").unwrap(),
+                &Mime::from_str("text/plain").unwrap(),
+            ),
             false
         );
-        assert_eq!(mime_db.mime_type_equal("/", "vnd/vnd"), false);
         assert_eq!(
-            mime_db.mime_type_equal("application/octet-stream", "text/plain"),
+            mime_db.mime_type_equal(
+                &Mime::from_str("text/plain").unwrap(),
+                &Mime::from_str("text/*").unwrap(),
+            ),
             false
         );
-        assert_eq!(mime_db.mime_type_equal("text/plain", "text/*"), false);
     }
 
+    // FIXME: Mime::from_str("/") yields an error?
+    // #[test]
+    // fn only_slash_is_vnd_vnd() {
```

Comment:
This should be `only_slash_isnt_vnd_vnd`, I guess.

Reply:
Oh, I see, I got it backwards.